azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -259,7 +259,6 @@ public void run() throws Exception {
         }
         throw new RuntimeException(e);
       } finally {
-        this.process = null;
```

Comment:
@suvodeep-pyne thanks for your input. I think we can simplify the life cycle further by having only 2 status: RUNNING, NOT_RUNNING and let jobrunner to take care of other status like killed/complete/failed.